### PR TITLE
fix: update pr workflow

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A- I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A- I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by keithwillcode — CHANGES_REQUESTED** [review]
We shouldn't be removing this. We don't want these builds to always run. They only need to run when we run the full test suite.

File: .github/workflows/publish-report.yml [hunk 1]
```diff
@@ -71,15 +71,15 @@ jobs:
...
+        uses: peter-evans/find-comment@v3.1.0
         id: fc
         with:
           issue-number: ${{ github.event.pull_request.number }}
           comment-author: "github-actions[bot]"
           body-includes: <!-- GENERATED-E2E-RESULTS -->
       - name: Create comment
         if: steps.fc.outputs.comment-id == ''
-        uses: peter-evans/create-or-update-comment@v3
+        uses: peter-evans/create-or-update-comment@v4.0.0
```
Comment (anikdhabal) [inline .github/workflows/publish-report.yml:82]: Use the latest version to fix the deprecation warning

Comment (anikdhabal) [reply .github/workflows/publish-report.yml:82]: cc @keithwillcode

File: .github/workflows/nextjs-bundle-analysis-annotation.yml [hunk 2]
```diff
@@ -57,22 +57,22 @@ jobs:
           echo ::set-output name=body::$body
 
       - name: Find Comment
-        uses: peter-evans/find-comment@v2
+        uses: peter-evans/find-comment@v3.1.0
```
Comment (keithwillcode) [inline .github/workflows/nextjs-bundle-analysis-annotation.yml:60]: I’m pretty sure the newer versions of this have data encoding issues. Let’s look and find an old PR I worked on to verify.

**Review by keithwillcode — CHANGES_REQUESTED** [review]

File: .github/workflows/nextjs-bundle-analysis-annotation.yml (hunk 2 above)
Comment (anikdhabal) [reply .github/workflows/nextjs-bundle-analysis-annotation.yml]: Hmm, also found issue in the latest version of /create-or-update-comment. Let's keep it as it is for now

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**